src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlServiceImpl.java

```diff
@@ -0,0 +1,161 @@
+package edu.cornell.kfs.module.purap.batch.service.impl;
+
+import java.io.IOException;
+import java.sql.Date;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.UUID;
+
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.collections4.ListUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+import org.kuali.kfs.core.api.datetime.DateTimeService;
+import org.kuali.kfs.sys.service.FileStorageService;
+
+import edu.cornell.kfs.module.purap.CUPurapConstants.JaggaerUploadSuppliersProcessingMode;
+import edu.cornell.kfs.module.purap.JaggaerConstants;
+import edu.cornell.kfs.module.purap.batch.service.JaggaerGenerateSupplierXmlService;
+import edu.cornell.kfs.module.purap.jaggaer.supplier.xml.Address;
+import edu.cornell.kfs.module.purap.jaggaer.supplier.xml.AddressList;
+import edu.cornell.kfs.module.purap.jaggaer.supplier.xml.Authentication;
+import edu.cornell.kfs.module.purap.jaggaer.supplier.xml.Header;
+import edu.cornell.kfs.module.purap.jaggaer.supplier.xml.JaggaerBuilder;
+import edu.cornell.kfs.module.purap.jaggaer.supplier.xml.Supplier;
+import edu.cornell.kfs.module.purap.jaggaer.supplier.xml.SupplierRequestMessage;
+import edu.cornell.kfs.module.purap.jaggaer.supplier.xml.SupplierSyncMessage;
+import edu.cornell.kfs.sys.CUKFSConstants;
+import edu.cornell.kfs.sys.service.CUMarshalService;
+import jakarta.xml.bind.JAXBException;
+
+public class JaggaerGenerateSupplierXmlServiceImpl implements JaggaerGenerateSupplierXmlService {
+    private static final Logger LOG = LogManager.getLogger();
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormat
+            .forPattern(CUKFSConstants.DATE_FORMAT_yyyy_MM_dd_T_HH_mm_ss_SSS_Z).withLocale(Locale.US);
```

Comment:
This particular date pattern is meant for UTC dates and also contains colons (`:`). I would recommend using `CUKFSConstants.DATE_FORMAT_yyyyMMdd_HHmmssSSS` instead, which should be more filename-friendly.

Reply:
OK